hazelcast/src/main/java/com/hazelcast/spi/discovery/DiscoveryStrategy.java

```diff
@@ -29,17 +32,18 @@
  * Using the simple lifecycle management strategies like multicast discovery is able to
  * register and destroy sockets based on Hazelcast’s lifecycle. Deactivated services will
  * also never be started.
+ *
+ * @since 3.6
  */
+@Beta
 public interface DiscoveryStrategy {
 
     /**
      * The <tt>start</tt> method is used to initialize internal state and perform any kind of
      * startup procedure like multicast socket creation. The behavior of this method might
-     * change based on the passed {@link DiscoveredNode}.
-     *
-     * @param discoveryMode the provided discovery mode
+     * change based on the {@link Member} instance passed to the {@link DiscoveryStrategyFactory}.
```

Comment:
I see that DefaultDiscoveryService has a reference to Member. 

You are only using it for this method:
```
   this.discoveryNode = buildDiscoveryNode(member);
```

where the following logic is executed:

```
   private DiscoveryNode buildDiscoveryNode(Member member) {
        return member == null ? null : new SimpleDiscoveryNode(member.getAddress(), member.getAttributes());
    }
```

I would just pass the argument into the constructor and remove any reference to member from this whole package.